keras_cv/layers/object_detection/non_max_suppression.py

```diff
@@ -88,8 +88,17 @@ def call(
 
         confidence_prediction = ops.max(class_prediction, axis=-1)
 
-        # TODO(tirthasheshpatel): Use backend-specific op where available
-        if multi_backend():
+        if not multi_backend() or keras.backend.backend() == "tensorflow":
+            idx, valid_det = tf.image.non_max_suppression_padded(
+                box_prediction,
+                confidence_prediction,
+                max_output_size=self.max_detections,
+                iou_threshold=self.iou_threshold,
+                score_threshold=self.confidence_threshold,
+                pad_to_max_output_size=True,
+                sorted_input=False,
+            )
+        elif keras.backend.backend() == "torch":
             # Since TorchVision has a nice efficient NMS op, we might as well
             # use it!
             if keras.backend.backend() == "torch":
```

Comment:
We are checking this condition twice now

Reply:
Welp that's embarrassing -- fixed